x-pack/filebeat/input/gcs/job.go

```diff
@@ -263,29 +304,32 @@ func (j *job) readJsonAndPublish(ctx context.Context, r io.Reader, id string) er
 func (j *job) publish(evt beat.Event, last bool, id string) {
 	if last {
 		// if this is the last object, then perform a complete state save
-		cp, done := j.state.saveForTx(j.object.Name, j.object.Updated)
+		cp, done := j.state.saveForTx(j.object.Name, j.object.Updated, j.metrics)
 		if err := j.publisher.Publish(evt, cp); err != nil {
+			j.metrics.errorsTotal.Inc()
 			j.log.Errorw("job encountered an error while publishing event", "gcs.jobId", id, "error", err)
 		}
 		done()
 		return
 	}
 	// since we don't update the cursor checkpoint, lack of a lock here is not a problem
 	if err := j.publisher.Publish(evt, nil); err != nil {
+		j.metrics.errorsTotal.Inc()
 		j.log.Errorw("job encountered an error while publishing event", "gcs.jobId", id, "error", err)
 	}
 }
 
 // splitEventList splits the event list into individual events and publishes them
-func (j *job) splitEventList(key string, raw json.RawMessage, offset int64, objHash string, id string) error {
+func (j *job) splitEventList(key string, raw json.RawMessage, offset int64, id string) (int, error) {
 	var jsonObject map[string]json.RawMessage
+	var evensPerObject int
```

Comment:
Typo?

Reply:
Yup, will fix it and make the suggested changes